03_Chatter/02-add-a-perisistence-layer.markdown

```diff
@@ -59,4 +62,86 @@ What this does is spawn an elixir process to do the save, outside of the request
 
 ### Display recent messages when a user joins the channel
 
-So, we have some messages saved, but it would be nice to see them when joining the channel. To do that, we have to change the `join/2` method to return a list of messages, and have the front end code parse and display those messages.
+So, we have some messages saved, but it would be nice to see them when joining the channel. To do that, we have to change the `join/3` method to return a list of messages, and have the front end code parse and display those messages.
+
+First, in order to complete the joining process, we have to send back `{:ok, socket}` from the `join/3` call. We can't send messages through the socket until the joining process has been completed. Instead, we'll send a call to the `gen_server` underlying our channel to send the messages _after_ the process has completed.
```

Comment:
maybe a sentence at the end of this paragraph about what `gen_server` is? Not necessarily an in depth explanation, but enough that it is not an abstract idea?